consensus/src/consensus/consensus_proxy.rs

```diff
@@ -235,4 +246,57 @@ impl<N: Network> ConsensusProxy<N> {
 
         Ok(transactions)
     }
+
+    pub async fn subscribe_transaction_addresses(
+        &self,
+        addresses: Vec<Address>,
+        min_peers: usize,
+        _max: Option<u16>,
+    ) {
+        // First we tell the network to provide us with a vector that contains all the connected peers that support such services
+        // Note: If the network could not provide enough peers that satisfies our requirement, then an error would be returned
+        let peers = self
+            .network
+            .get_peers_by_services(Services::MEMPOOL, min_peers)
+            .await
+            .map_err(|error| {
+                log::error!(
+                    err = %error,
+                    "The transactions by address request couldn't be fulfilled"
+                );
+
+                RequestError::OutboundRequest(OutboundRequestError::SendError)
+            });
+        // Subscribe to any peer that could provide the necessary services
+        for peer_id in peers.unwrap() {
+            let response = self
+                .network
+                .request::<RequestSubscribeToAddress>(
+                    RequestSubscribeToAddress {
+                        operation: TxAddressSubscriptionOperation::Initiate,
```

Comment:
Here the `::Initiate` type is used. The other types `::Update` and `::Terminate` are not used (yet). Do you have a plan to use those, too?